internal/githubapi/errors_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2022 Criticality Score Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package githubapi
+
+import (
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-github/v47/github"
+)
+
+func TestErrorResponseStatusCode(t *testing.T) {
+	tests := []struct { //nolint:govet
+		name string
+		err  error
+		want int
+	}{
+		{
+			name: "nil error",
+			want: 0,
+		},
+		{
+			name: "non-nil error that is not in ErrorResponse",
+			err:  errors.New("some error"),
+			want: 0,
+		},
+		{
+			name: "error that is in ErrorResponse",
+			err: &github.ErrorResponse{
+				Response: &http.Response{
+					StatusCode: 404,
+				},
+			},
+			want: 404,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := ErrorResponseStatusCode(test.err); got != test.want {
+				t.Errorf("ErrorResponseStatusCode() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+func TestGraphQLErrors_Error(t *testing.T) {
+	tests := []struct { //nolint:govet
+		name      string
+		errors    []GraphQLError
+		want      string
+		wantPanic bool
+	}{
+		{
+			name:      "zero errors",
+			errors:    []GraphQLError{},
+			wantPanic: true,
+		},
+		{
+			name: "one error",
+			errors: []GraphQLError{
+				{Message: "one"},
+			},
+			want: "one",
+		},
+		{
+			name: "more than one error",
+			errors: []GraphQLError{
+				{Message: "one"},
+				{Message: "two"},
+				{Message: "three"},
+			},
+			want: "3 GraphQL errors",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			defer func() {
+				r := recover()
+				if r != nil && !test.wantPanic {
+					t.Fatalf("Error() panic: %v, %s", r, test.name)
+				}
+			}()
+
+			e := &GraphQLErrors{test.errors}
+
+			if got := e.Error(); got != test.want {
+				t.Errorf("Error() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+func TestGraphQLErrors_HasType(t *testing.T) {
+	tests := []struct { //nolint:govet
+		name   string
+		errors []GraphQLError
+		t      string
+		want   bool
+	}{
```

Comment:
Please add two more tests:

1. two GraphQLErrors one with NOT_FOUND, one without.
2. two GraphQLErrors where their Type fields are set to something other than the search.

Reply:
I have added the two tests.